cookbook/workflows/README.md

```diff
@@ -0,0 +1,499 @@
+- Maximum flexibility
+- Custom logic integration
+- Component reusability
+
+**See**: [`sequence_of_functions_and_agents.py`](sync/sequence_of_functions_and_agents.py)
+
+### 7. Function-Based Steps
+
+**When to use**: Custom business logic, API integrations, data transformations.
+
+**Example**: Custom processing with agent integration
+
+![Custom Function Steps](/cookbook/workflows/assets/custom_function_steps.png)
+
+```python
+from agno.workflow.v2 import Step, Workflow
+
+def custom_processor(step_input) -> StepOutput:
+    # Custom logic with agent interaction
+    processed_data = some_custom_logic(step_input.message)
+    agent_response = content_agent.run(processed_data)
+    
+    return StepOutput(
+        content=f"Enhanced: {agent_response.content}",
+        response=agent_response
+    )
+
+workflow = Workflow(
+    name="Custom Processing Pipeline",
+    steps=[
+        Step(name="Research", agent=researcher),
+        Step(name="Custom Processing", executor=custom_processor),
+        Step(name="Final Review", agent=reviewer),
+    ]
+)
+```
+
+**Key Benefits**:
+- Custom business logic
+- External API integration
+- Data transformation capabilities
+
+**See**: [`step_with_function.py`](sync/step_with_function.py)
+
+### 8. Complex Combinations
+
+**When to use**: Sophisticated workflows requiring multiple patterns.
+
+**Example**: Conditions + Parallel + Loops + Routing
+
+```python
+from agno.workflow.v2 import Condition, Loop, Parallel, Router, Step, Workflow
+
+# Complex workflow combining multiple patterns
+workflow = Workflow(
+    name="Advanced Multi-Pattern Workflow",
+    steps=[
+        Parallel(
+            Condition(
+                name="Tech Check",
+                evaluator=is_tech_topic,
+                steps=[Step(name="Tech Research", agent=tech_researcher)]
+            ),
+            Condition(
+                name="Business Check", 
+                evaluator=is_business_topic,
+                steps=[
+                    Loop(
+                        name="Deep Business Research",
+                        steps=[Step(name="Market Research", agent=market_researcher)],
+                        end_condition=research_quality_check,
+                        max_iterations=3
+                    )
+                ]
+            ),
+            name="Conditional Research Phase"
+        ),
+        Router(
+            name="Content Type Router",
+            selector=content_type_selector,
+            choices=[blog_post_step, social_media_step, report_step]
+        ),
+        Step(name="Final Review", agent=reviewer),
+    ]
+)
+```
+
+**Key Benefits**:
+- Ultimate flexibility
+- Complex business logic
+- Scalable architecture
+
+**See**: [`condition_and_parallel_steps.py`](sync/condition_and_parallel_steps.py), [`router_with_loop_steps.py`](sync/router_with_loop_steps.py)
+
+## Advanced Features
+
+### Streaming Support
+
+Enable real-time processing for long-running workflows:
```

Comment:
it is always realtime. This just means events are streaming, which is the real benefit.